Abot2.Tests.Unit/Core/HyperlinkParserTest.cs

```diff
@@ -255,67 +253,91 @@ public void GetLinks_WhiteSpaceHtml()
+                DisplayName = "Terminated absolute base tag (no path)"
+            ),
+            DataRow(
+                "http://bbb.com",
+                HTTP_URI_STRING,
+                "http://aaa.com/", "http://bbb.com/aaa/a.html",
+                DisplayName = "Unterminated absolute base tag (no path)"
+            ),
+            DataRow(
+                "http://bbb.com/images/",
+                HTTP_URI_STRING,
+                "http://aaa.com/", "http://bbb.com/aaa/a.html", "http://bbb.com/images/aaa/a.html",
+                DisplayName = "Terminated absolute base tag (with path)"
+            ),
+            DataRow(
+                "http://bbb.com/images",
+                HTTP_URI_STRING,
+                "http://aaa.com/", "http://bbb.com/aaa/a.html",
+                DisplayName = "Unterminated absolute base tag (with path)"
+            ),
+            DataRow(
+                "/images/",
+                HTTP_URI_STRING,
+                "http://aaa.com/", "http://a.com/aaa/a.html", "http://a.com/images/aaa/a.html",
+                DisplayName = "Terminated root-relative base tag"
+            ),
+            DataRow(
+                "/images",
+                HTTP_URI_STRING,
+                "http://aaa.com/", "http://a.com/aaa/a.html",
+                DisplayName = "Unterminated root-relative base tag"
+            ),
+            DataRow(
+                "images/",
+                HTTP_URI_STRING,
+                "http://aaa.com/", "http://a.com/aaa/a.html", "http://a.com/images/aaa/a.html",
+                DisplayName = "Terminated relative base tag"
+            ),
+            DataRow(
+                "images",
+                HTTP_URI_STRING,
+                "http://aaa.com/", "http://a.com/aaa/a.html",
+                DisplayName = "Unterminated relative base tag"
+            ),
+            DataRow(
+                "////images/",
+                HTTP_URI_STRING,
+                "http://aaa.com/", "http://a.com/aaa/a.html",
+                DisplayName = "Invalid base tag"
+            )
+        ]
+        public void GetLinks_BaseTagPresent_ReturnsExpectedLinks(string baseTagHref, params string[] moreData)
+        {
+            var parentPageUri = new Uri(moreData[0]);
+            List<string> expectedLinks = moreData.Skip(1).ToList();
+
+            _crawledPage = CreateCrawledPage(parentPageUri);
+            _crawledPage.Content.Text = $"<base href=\"{baseTagHref}\"><a href=\"http://aaa.com/\"></a><a href=\"/aaa/a.html\"></a><a href=\"aaa/a.html\"></a>";
+            List<string> actualLinks = _unitUnderTest.GetLinks(_crawledPage).Select(x => x.HrefValue.AbsoluteUri).ToList();
+
+            Assert.AreEqual(expectedLinks.Count, actualLinks.Count, "Unexpected amount of links returned");
+            for (int i = 0; i < expectedLinks.Count; i++)
+            {
+                Assert.AreEqual(expectedLinks[i], actualLinks[i]);
+            }
         }
 
-        [TestMethod]
```

Comment:
Please do not delete ANY tests and make your test follow the same naming convention. Even if it's a little more ceremony.

Reply:
To my knowledge, the consolidated test did follow the naming convention. The DisplayName is supposed to just be the name of the individual outcome of that test, but I'm not familiar with the testing library being used here so I could be wrong.